lib/proxy/plugin_initializer.rb

```diff
@@ -338,7 +338,7 @@ def load_classes
   end
 
   def validate_settings(plugin, config)
-    result = execute_validators(plugin.plugin_default_settings.keys.map { |k| {:name => :presence, :setting => k} }, config)
+    result = execute_validators(plugin.plugin_default_settings.keys.map { |k| {:name => :presence_allow_empty, :setting => k} }, config)
```

Comment:
This line is my concern. My reading of this is that it's changing the validator for every setting with a specified validator from `presence` to `presence_allow_empty`. That means suddenly **all** settings with a default are allowed to be empty strings. The alternative would be to remove the default value just for `dns_server` and rely on the installer to set the correct default. That would be a much more minimal patch since the `PresenceAllowEmpty` validator is no longer needed.